ts/indicators/ema.src.ts

```diff
@@ -0,0 +1,230 @@
+            // for inheritance
+        }
+
+        class EMAIndicatorPoint extends SMAIndicatorPoint {
+            public series: EMAIndicator;
+        }
+
+        interface SeriesTypesDictionary {
+            ema: typeof EMAIndicator;
+        }
+
+    }
+}
+
+import U from '../parts/Utilities.js';
+var isArray = U.isArray;
+
+var seriesType = H.seriesType,
+    correctFloat = H.correctFloat;
+
+/**
+ * The EMA series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.ema
+ *
+ * @augments Highcharts.Series
+ */
+seriesType<Highcharts.EMAIndicator>(
+    'ema',
+    'sma',
+    /**
+     * Exponential moving average indicator (EMA). This series requires the
+     * `linkedTo` option to be set.
+     *
+     * @sample stock/indicators/ema
+     *         Exponential moving average indicator
+     *
+     * @extends      plotOptions.sma
+     * @since        6.0.0
+     * @product      highstock
+     * @optionparent plotOptions.ema
+     */
+    {
+        params: {
+            /**
+             * The point index which indicator calculations will base. For
+             * example using OHLC data, index=2 means the indicator will be
+             * calculated using Low values.
+             *
+             * By default index value used to be set to 0. Since Highstock 7
+             * by default index is set to 3 which means that the ema
+             * indicator will be calculated using Close values.
+             */
+            index: 3,
+            period: 9 // @merge 14 in v6.2
+        }
+    },
+    /**
+     * @lends Highcharts.Series#
+     */
+    {
+        accumulatePeriodPoints: function (
+            period: number,
+            index: number,
+            yVal: Array<Array<number>>
+        ): number {
+            var sum = 0,
+                i = 0,
+                y: (number|Array<number>) = 0;
+
+            while (i < period) {
+                y = index < 0 ? yVal[i] : yVal[i][index];
+                sum = sum + (y as any);
+                i++;
+            }
+
+            return sum;
+        },
+        calculateEma: function (
+            xVal: (Array<number>|undefined),
+            yVal: Array<Array<number>>,
+            i: number,
+            EMApercent: number,
+            calEMA: (number|undefined),
+            index: number,
+            SMA: number
+        ): [number, number] {
+            var x: number = (xVal as any)[i - 1],
+                yValue: (number|Array<number>) = index < 0 ?
+                    yVal[i - 1] :
+                    yVal[i - 1][index],
+                y: number;
+
+            y = calEMA === undefined ?
+                SMA : correctFloat(((yValue as any) * EMApercent) +
+                (calEMA * (1 - EMApercent)));
+
+            return [x, y];
+        },
+        getValues: function (
+            this: Highcharts.EMAIndicator,
+            series: Highcharts.EMAIndicator,
```

Comment:
Here the `series` parameters is about the linked parent series. Therefore type `Highcharts.Series` is correct. At the same time you do not need `yData` on the indicator class, so it can be removed in the declaration.

Reply:
Ok, I have made some changes, but this still confuses me in the sense that we previously moved the `yData` in order to be able to define it in the indicator. So, that change was perhaps not necessary given that it is the linkedParent that is passed to getValues?

Also given the above, I think the same fix needs to be applied in other indicators as well.

Reply:
That is then probably incorrect.`yData` on the indicator is not in direct usage as far as I know.

The `series` parameter of `Indicator.getValues` is always about the linked parent series, not about the indicator.

We created in CMF an interface for OHLC-like series with a two-dimensional `yData` property to cover any compatible series type: https://github.com/highcharts/highcharts/blob/master/ts/indicators/cmf.src.ts#L62-L64

Reply:
I have now changed the Series type in the getValues(..) implementation of several indicators. Please review @bre1470 :)

Reply:
This one is solved. 👍